toolchain/check/import_ref.cpp

```diff
@@ -125,44 +125,58 @@ auto VerifySameCanonicalImportIRInst(Context& context, SemIR::InstId prev_id,
 // Calling Resolve on an instruction operates in an iterative manner, tracking
 // Work items on work_stack_. At a high level, the loop is:
 //
-// 1. If a constant value is already known for the work item, and we're
-//    processing it for the first time, it's considered resolved.
+// 1. If a constant value is already known for the work item and was not set by
+//    this work item, it's considered resolved.
 //    - The constant check avoids performance costs of deduplication on add.
-//    - If `retry` is set, we process it again, because it didn't complete last
-//      time, even though we have a constant value already.
-// 2. Resolve the instruction: (TryResolveInst/TryResolveTypedInst)
-//    - For a symbolic constant within a generic, find the generic itself. If it
-//      needs to be imported, return Retry() to import the generic before we
-//      import the constant.
-//    - For instructions that can be forward declared, if we don't already have
-//      a constant value from a previous attempt at resolution, start by making
-//      a forward declared constant value to address circular references.
-//    - Gather all input constants.
+//    - If we've processed this work item before, then we now process it again,
+//      because it didn't complete last time, even though we have a constant
+//      value already.
+//
+// 2. Resolve the instruction (TryResolveInst/TryResolveTypedInst), in up to
+//    three passes. First pass:
```

Comment:
I wonder, is  "pass" the right phrasing here? I think these *can* be a pass, but maybe sub-step phrasing would work better. e.g.:

```suggestion
// 2. Resolve the instruction (TryResolveInst/TryResolveTypedInst). This is done in three sub-steps, each of which can result end in a retry. Whenever one does end in a retry, it will be rerun but always proceed to the next sub-step (up to three passes total, one per sub-step). First sub-step:
```

(note, other pass -> sub-step remaining would probably be appropriate)

Reply:
I think for the purpose of cross-referencing this from other comments it's useful to use a different word than "step" / "sub-step" (given that we also cross-reference the steps). I've gone with "phase" for now.